lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -723,8 +730,116 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  defp do_accept_transaction(%Transaction{type: :contract, data: %TransactionData{code: code}}, _) do
+    cond do
+      byte_size(code) == 0 ->
+        {:error, "invalid contract type transaction -  code is empty "}
+
+      byte_size(code) >= @code_max_size ->
+        {:error, "invalid contract type transaction , code exceed max size "}
+
+      true ->
+        :ok
+    end
+  end
+
+  defp do_accept_transaction(
+         %Transaction{
+           type: :data,
+           data: data = %TransactionData{content: content, ownerships: ownerships},
+           version: tx_version
+         },
+         _
+       ) do
+    nb_ownerships = length(ownerships)
+
+    tx_data_size =
+      data
+      |> TransactionData.serialize(tx_version)
+      |> :erlang.byte_size()
```

Comment:
I think you can use `byte_size` directly

Reply:
But the limitation of the size should be global, not only for this type of transaction